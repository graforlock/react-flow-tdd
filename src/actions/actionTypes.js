export const FETCH_GET_ALL_IDEAS = 'FETCH_GET_ALL_IDEAS'
export const FETCH_GET_NEW_IDEA = 'FETCH_GET_NEW_IDEA'
export const FETCH_UPDATE_IDEA = 'FETCH_UPDATE_IDEA'
export const FETCH_DELETE_IDEA = 'FETCH_DELETE_IDEA'

export const RECEIVE_ALL_IDEAS = 'RECEIVE_ALL_IDEAS'
export const RECEIVE_NEW_IDEA = 'RECEIVE_NEW_IDEA'

type IdeasFetchAction = {
  type:
  FETCH_GET_ALL_IDEAS |
  FETCH_GET_NEW_IDEA |
  FETCH_UPDATE_IDEA |
  FETCH_DELETE_IDEA,
  payload: ?number
}

type IdeasReceiveAction = {
  type:
  RECEIVE_ALL_IDEAS |
  RECEIVE_NEW_IDEA,
  payload: Array<Object>
}

export type { IdeasFetchAction, IdeasReceiveAction }
