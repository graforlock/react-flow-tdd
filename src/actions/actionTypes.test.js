// @flow
import * as actionTypes from './actionTypes'

describe('actionTypes', () => {
  it('matches the current actionTypes snapshot', () => {
    expect(actionTypes).toMatchSnapshot()
  })
})
