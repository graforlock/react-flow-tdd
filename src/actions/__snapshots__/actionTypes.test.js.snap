// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actionTypes matches the current actionTypes snapshot 1`] = `
Object {
  "EDITOR_DISABLE_WRITING": "EDITOR_DISABLE_WRITING",
  "EDITOR_EDIT_BODY": "EDITOR_EDIT_BODY",
  "EDITOR_EDIT_CLEAR": "EDITOR_EDIT_CLEAR",
  "EDITOR_EDIT_TITLE": "EDITOR_EDIT_TITLE",
  "EDITOR_ENABLE_WRITING": "EDITOR_ENABLE_WRITING",
  "FETCH_CREATE_IDEA": "FETCH_CREATE_IDEA",
  "FETCH_DELETE_IDEA": "FETCH_DELETE_IDEA",
  "FETCH_GET_ALL_IDEAS": "FETCH_GET_ALL_IDEAS",
  "FETCH_UPDATE_IDEA": "FETCH_UPDATE_IDEA",
  "RECEIVE_ALL_IDEAS": "RECEIVE_ALL_IDEAS",
  "RECEIVE_DELETED_IDEA": "RECEIVE_DELETED_IDEA",
  "RECEIVE_NEW_IDEA": "RECEIVE_NEW_IDEA",
  "RECEIVE_UPDATED_IDEA": "RECEIVE_UPDATED_IDEA",
  "REQUEST_CREATE_FAILURE": "REQUEST_CREATE_FAILURE",
  "REQUEST_CREATE_SUCCESS": "REQUEST_CREATE_SUCCESS",
  "REQUEST_DELETE_FAILURE": "REQUEST_DELETE_FAILURE",
  "REQUEST_DELETE_SUCCESS": "REQUEST_DELETE_SUCCESS",
  "REQUEST_GET_ALL_FAILURE": "REQUEST_GET_ALL_FAILURE",
  "REQUEST_GET_ALL_SUCCESS": "REQUEST_GET_ALL_SUCCESS",
  "REQUEST_UPDATE_FAILURE": "REQUEST_UPDATE_FAILURE",
  "REQUEST_UPDATE_SUCCESS": "REQUEST_UPDATE_SUCCESS",
}
`;
